File: src/contests/cg/puzzle/php/distributing-candy.php
<?php
// 🎮 CodinGame Puzzle - distributing-candy
// https://www.codingame.com/training/easy/distributing-candy

list($n, $m) = array_map('intval', explode(' ', trim(fgets(STDIN))));
$candies = array_map('intval', explode(' ', trim(fgets(STDIN))));
sort($candies);
$best = PHP_INT_MAX;
for ($i = 0; $i + $m - 1 < $n; $i++) {
    $diff = $candies[$i + $m - 1] - $candies[$i];
    if ($diff < $best) {
        $best = $diff;
    }
}
echo $best . "\n";

File: src/contests/cg/puzzle/cpp/distributing-candy.cpp
// 🎮 CodinGame Puzzle - distributing-candy
// https://www.codingame.com/training/easy/distributing-candy

#include <iostream>
#include <vector>
#include <algorithm>
#include <climits>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    vector<int> c(n);
    for (int i = 0; i < n; i++) cin >> c[i];
    sort(c.begin(), c.end());
    long long best = LLONG_MAX;
    for (int i = 0; i + m - 1 < n; i++) {
        int diff = c[i + m - 1] - c[i];
        if (diff < best) best = diff;
    }
    cout << best << endl;
}
